src/overlays/actors/ovl_Bg_F40_Swlift/z_bg_f40_swlift.c

```diff
@@ -28,24 +31,81 @@ ActorInit Bg_F40_Swlift_InitVars = {
     (ActorFunc)BgF40Swlift_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_8096F540[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneScale, 550, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 5000, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+void BgF40Swlift_Init(Actor* thisx, PlayState* play) {
+    s32 index;
+    BgF40Swlift* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init((DynaPolyActor*)this, 1);
+    index = BG_F40_GET_SWITCHFLAG(this);
+    if ((index < 0) || (index >= 5)) { //! @bug An index greater than 3 will cause an out of bounds array access.
+        Actor_Kill(&this->dyna.actor);
+    } else {
+        sHeights[index] = this->dyna.actor.world.pos.y;
+        sParams[index] = BG_F40_SWLIFT_GET_PARAMS(this);
```

Comment:
Think you got the switch flags backwards, the switch flag gets passed to the Flags_GetSwitch and SetSwitch functions, in this case, index appears to be unique identifier, to help keep multiple actors separated from each other (seems unnecessary unless they, or some other actor, would try to change the actors values) GetParams saves the new switch flag to the static array, which gets read later.

Reply:
Would BG_F40_GET_IDENTIFIER work as a name? Sorry for the very long delay.

Reply:
I would suggest just `BG_F40_GET_START_INDEX`. Based on Update it cycles through different indices, and this is just the one it starts with.

Reply:
Got it after finding an example in the code. Will name GET_PARAMS to GET_SWITCHFLAG and GET_SWITCHFLAG to GET_INDEX.